lib/internal/test_runner/runner.js

```diff
@@ -112,47 +118,145 @@ function getRunArgs({ path, inspectPort }) {
   return argv;
 }
 
+class FileTest extends Test {
+  #buffer = [];
+  #handleReportItem({ kind, node, nesting = 0 }) {
+    const indent = StringPrototypeRepeat(kDefaultIndent, nesting + 1);
+
+    const details = (diagnostic) => {
+      return (
+        diagnostic && {
+          __proto__: null,
+          yaml:
+            `${indent}  ` +
+            ArrayPrototypeJoin(diagnostic, `\n${indent}  `) +
+            '\n',
+        }
+      );
+    };
+
+    switch (kind) {
+      case TokenKind.TAP_VERSION:
+        // TODO(manekinekko): handle TAP version
+        // this.reporter.version(node.version);
+        break;
+
+      case TokenKind.TAP_PLAN:
+        this.reporter.plan(indent, node.end - node.start + 1);
+        break;
+
+      case TokenKind.TAP_SUBTEST_POINT:
+        this.reporter.subtest(indent, node.name);
+        break;
+
+      case TokenKind.TAP_TEST_OK:
+      case TokenKind.TAP_TEST_NOTOK:
+        // eslint-disable-next-line no-case-declarations
+        const { todo, skip, pass } = node.status;
+        // eslint-disable-next-line no-case-declarations
+        let directive = {};
+
+        if (todo) {
+          directive = this.reporter.getTodo(node.reason);
+        }
+
+        if (skip) {
+          directive = this.reporter.getSkip(node.reason);
+        }
+
+        if (pass) {
+          this.reporter.ok(
+            indent,
+            node.id,
+            node.description,
+            null,
+            directive
+          );
+        } else {
+          this.reporter.fail(
+            indent,
+            node.id,
+            node.description,
+            null,
+            directive
+          );
+        }
+        break;
+
+      case TokenKind.TAP_YAML_START:
+        break; // skip
+
+      case TokenKind.TAP_YAML_END:
+        this.reporter.details(indent, details(node.diagnostics));
```

Comment:
I would expect the parser to emit the yaml details together with the test itself

Reply:
Fixed in the parser. I need to update the runner.

Reply:
fixed.